src/transaction/log_page_buffer.c

```diff
@@ -3988,12 +3989,22 @@ logpb_flush_all_append_pages (THREAD_ENTRY * thread_p)
 	  error_code = ER_FAILED;
 	  goto error;
 	}
-      tmp_eof = (LOG_RECORD_HEADER *) ((char *) first_append_log_page->area + log_Gl.append.prev_lsa.offset);
+
+      copy_to_first_append = (LOG_PAGE *) malloc (LOG_PAGESIZE);
+      memcpy (copy_to_first_append, first_append_log_page, LOG_PAGESIZE);
+      first_append_pageid = log_Gl.append.prev_lsa.pageid;
+      tmp_eof = (LOG_RECORD_HEADER *) ((char *) copy_to_first_append->area + log_Gl.append.prev_lsa.offset);
       save_record = *tmp_eof;
 
       /* Overwrite it with an end of log marker */
       LSA_SET_NULL (&tmp_eof->forw_lsa);
       tmp_eof->type = LOG_END_OF_LOG;
+      if (logpb_write_page_to_disk (thread_p, copy_to_first_append, first_append_pageid) != NO_ERROR)
+	{
+	  error_code = ER_FAILED;
+	  goto error;
+	}
+      logpb_set_dirty (thread_p, first_append_log_page);
```

Comment:
Actually we need to clear dirty flag here. first_append_log_page is most likely already dirty and in flush list. We want to avoid to flush it again to disk until the end of flush. Therefore, we should clear the log buffer dirty flag here.